src/core/shape_inference/include/squeeze_shape_inference.hpp

```diff
@@ -50,11 +50,34 @@ void shape_infer(const Squeeze* op,
         OPENVINO_SUPPRESS_DEPRECATED_END
 
         std::vector<int64_t> axes;
-        if (arg_rank.is_static() && axes_shape.is_static() && get_data_as_int64<T>(1, op, axes, constant_data)) {
-            OPENVINO_SUPPRESS_DEPRECATED_START
-            normalize_axes(op, arg_rank.get_length(), axes);
-            OPENVINO_SUPPRESS_DEPRECATED_END
-            unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+        if (arg_rank.is_static() && axes_shape.is_static()) {
+            if (get_data_as_int64<T>(1, op, axes, constant_data)) {
+                OPENVINO_SUPPRESS_DEPRECATED_START
+                normalize_axes(op, arg_rank.get_length(), axes);
+                OPENVINO_SUPPRESS_DEPRECATED_END
+                unique_axes.reset(new std::set<int64_t>(axes.cbegin(), axes.cend()));
+            } else if (axes_shape.size() == 1 && axes_shape[0].get_min_length() > 0) {
+                // The `axes` input is a Parameter
+                output_shape.resize(0);
+                int64_t squeezable_dims_count =
+                    std::count_if(arg_shape.begin(), arg_shape.end(), [&](const DimType& dim) {
+                        if (dim.compatible(1)) {
+                            return true;
+                        } else {
+                            // Copy not squeezable dimensions to the output shape
+                            output_shape.push_back(dim);
+                            return false;
+                        }
+                    });
+                if (squeezable_dims_count > axes_shape[0].get_min_length()) {
```

Comment:
Unfortunately, I cannot agree on this solution.

The following case output rank must be dynamic:
```

Parameter([-1, -1, 2, 3, -1, 5])---\
                                             Squeeze 
Parameter([2]) ------------------/

```

TensorFlow example:

![image](https://user-images.githubusercontent.com/55839243/236755312-d9f23927-36d5-4020-824d-9d4b2d4bc4ab.png)

I propose to exclusively work with scalars and 1D tensors with 1 element. We are sure that they are unique so we can build our solution on top of that.

Reply:
The logic for constant duplicated `axes` stays the same (as in Tensorflow example), 
the proposed change affects only `axes` as Parameter. 
Enablement of (non single element) Parameter as input is possible only by assuming uniqueness of axes, 
otherwise plugins are not able to infer dynamic rank anyway. 
So the hard choice is between keeping unsupported dynamic rank or accepting the risk of output shape mismatch (if axes are duplicated) in favour of extending Parameter input support.

For the agreement, I have updated the changes to support only Parameter with single element.